kernel/interrupt_controller/src/aarch64.rs

```diff
@@ -69,7 +72,10 @@ pub struct SystemInterruptController(IrqSafeMutex<ArmGicDistributor>);
 /// Struct representing per-cpu-core interrupt controller chips.
 ///
 /// On aarch64 w/ GIC, this corresponds to a Redistributor & CPU interface.
-pub struct LocalInterruptController(IrqSafeMutex<ArmGicCpuComponents>);
+pub struct LocalInterruptController(UnsafeCell<IrqSafeMutex<ArmGicCpuComponents>>);
```

Comment:
yeah, it's a tough call, as I'm not necessarily sure that `force_unlock` is actually better. I think you may be right to just completely circumvent the mutex entirely rather than attempt to modify its locked state at all (e.g., by force unlocking it). So i think we'll stick with your approach for now.